TEZ-4554: Counter for used nodes within a DAG

abstractdog opened the pull request:
tested on a small tez container mode cluster

with 2 healthy and running yarn nodemanagers:
```
INFO  : org.apache.tez.common.counters.DAGCounter:
...
INFO  :    NODE_USED_COUNT: 2
INFO  :    NODE_TOTAL_COUNT: 2
```

after stopping 1 nodemanager:
```
INFO  : org.apache.tez.common.counters.DAGCounter:
...
INFO  :    NODE_USED_COUNT: 1
INFO  :    NODE_TOTAL_COUNT: 2
```

after decommissioning 1 nodemanager:
```
INFO  : org.apache.tez.common.counters.DAGCounter:
...
INFO  :    NODE_USED_COUNT: 1
INFO  :    NODE_TOTAL_COUNT: 1
```


also tested with LLAP on Cloudera CDW (3 LLAP daemons)
```
INFO  :    NODE_USED_COUNT: 3
INFO  :    NODE_TOTAL_COUNT: 3
```

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  34m 24s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 45s |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |   0m 40s |  root in master failed.  |
| -1 :x: |  compile  |   0m 19s |  tez-api in master failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  compile  |   2m  3s |  tez-dag in master failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  compile  |   0m 19s |  tez-api in master failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -1 :x: |  compile  |   0m 19s |  tez-dag in master failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -0 :warning: |  checkstyle  |   0m 16s |  The patch fails to run checkstyle in tez-dag  |
| -1 :x: |  javadoc  |   0m 19s |  tez-api in master failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  javadoc  |   0m 18s |  tez-dag in master failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  javadoc  |   0m 18s |  tez-api in master failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -1 :x: |  javadoc  |   0m 19s |  tez-dag in master failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| +0 :ok: |  spotbugs  |   5m 49s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| -1 :x: |  findbugs  |   0m 19s |  tez-api in master failed.  |
| -1 :x: |  findbugs  |   0m 19s |  tez-dag in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 19s |  tez-api in the patch failed.  |
| -1 :x: |  mvninstall  |   0m 18s |  tez-dag in the patch failed.  |
| -1 :x: |  compile  |   0m 18s |  tez-api in the patch failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  compile  |   0m 18s |  tez-dag in the patch failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  javac  |   0m 18s |  tez-api in the patch failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  javac  |   0m 18s |  tez-dag in the patch failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  compile  |   0m 20s |  tez-api in the patch failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -1 :x: |  compile  |   0m 18s |  tez-dag in the patch failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -1 :x: |  javac  |   0m 20s |  tez-api in the patch failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -1 :x: |  javac  |   0m 18s |  tez-dag in the patch failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -0 :warning: |  checkstyle  |   0m 17s |  The patch fails to run checkstyle in tez-api  |
| -0 :warning: |  checkstyle  |   0m 17s |  The patch fails to run checkstyle in tez-dag  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  javadoc  |   0m 18s |  tez-api in the patch failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  javadoc  |   0m 20s |  tez-dag in the patch failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  javadoc  |   0m 19s |  tez-api in the patch failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -1 :x: |  javadoc  |   0m 19s |  tez-dag in the patch failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -1 :x: |  findbugs  |   0m 18s |  tez-api in the patch failed.  |
| -1 :x: |  findbugs  |   0m 19s |  tez-dag in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 19s |  tez-api in the patch failed.  |
| -1 :x: |  unit  |   0m 19s |  tez-dag in the patch failed.  |
| +0 :ok: |  asflicense  |   0m 38s |  ASF License check generated no output?  |
|  |   |  52m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/362 |
| JIRA Issue | TEZ-4554 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 9b422e391e8c 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / e08d0279c |
| Default Java | Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-compile-tez-api-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-compile-tez-dag-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-compile-tez-api-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-compile-tez-dag-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/buildtool-branch-checkstyle-tez-dag.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-javadoc-tez-api-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-javadoc-tez-dag-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-javadoc-tez-api-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-javadoc-tez-dag-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-findbugs-tez-api.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/branch-findbugs-tez-dag.txt |
| mvninstall | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-mvninstall-tez-api.txt |
| mvninstall | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-mvninstall-tez-dag.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-compile-tez-api-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-compile-tez-dag-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| javac | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-compile-tez-api-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| javac | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-compile-tez-dag-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-compile-tez-api-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-compile-tez-dag-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| javac | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-compile-tez-api-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| javac | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-compile-tez-dag-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/buildtool-patch-checkstyle-tez-api.txt |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/buildtool-patch-checkstyle-tez-dag.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-javadoc-tez-api-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-javadoc-tez-dag-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-javadoc-tez-api-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-javadoc-tez-dag-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-findbugs-tez-api.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-findbugs-tez-dag.txt |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-unit-tez-api.txt |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/artifact/out/patch-unit-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/testReport/ |
| Max. process+thread count | 46 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/2/console |
| versions | git=2.34.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  26m 25s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 36s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m 18s |  master passed  |
| +1 :green_heart: |  compile  |   1m 23s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   1m 14s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  checkstyle  |   1m 26s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 10s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   1m  1s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +0 :ok: |  spotbugs  |   1m 27s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  7s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 59s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  javac  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  findbugs  |   2m 26s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  17m 18s |  tez-api in the patch failed.  |
| -1 :x: |  unit  |   0m 34s |  tez-dag in the patch failed.  |
| +0 :ok: |  asflicense  |   0m 38s |  ASF License check generated no output?  |
|  |   |  79m 56s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | tez.common.TestTezCommonUtils |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/362 |
| JIRA Issue | TEZ-4554 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux fe3df60df3e5 5.15.0-106-generic #116-Ubuntu SMP Wed Apr 17 09:17:56 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / e08d0279c |
| Default Java | Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/1/artifact/out/patch-unit-tez-api.txt |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/1/artifact/out/patch-unit-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/1/testReport/ |
| Max. process+thread count | 581 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/1/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  28m  4s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m  4s |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |   2m 12s |  root in master failed.  |
| -1 :x: |  compile  |   0m 18s |  tez-api in master failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  compile  |   1m 12s |  tez-dag in master failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  compile  |   0m 18s |  tez-api in master failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -1 :x: |  compile  |   0m 19s |  tez-dag in master failed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08.  |
| -0 :warning: |  checkstyle  |   0m 16s |  The patch fails to run checkstyle in tez-api  |
| -0 :warning: |  checkstyle  |   0m 17s |  The patch fails to run checkstyle in tez-dag  |
| -1 :x: |  javadoc  |   0m 18s |  tez-api in master failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| -1 :x: |  javadoc  |   1m 14s |  tez-dag in master failed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.  |
| +1 :green_heart: |  javadoc  |   0m 59s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +0 :ok: |  spotbugs  |   3m 34s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| -1 :x: |  findbugs  |   0m  8s |  tez-api in master failed.  |
| -1 :x: |  findbugs  |   0m 19s |  tez-dag in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 20s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  2s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 51s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  javac  |   0m 51s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 24s |  tez-dag: The patch generated 176 new + 0 unchanged - 0 fixed = 176 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  javadoc  |   0m 25s |  tez-api-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 generated 13 new + 0 unchanged - 0 fixed = 13 total (was 0)  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  findbugs  |   3m 16s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 24s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 22s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 25s |  The patch does not generate ASF License warnings.  |
|  |   |  61m 19s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/362 |
| JIRA Issue | TEZ-4554 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux ee76df0059a8 5.15.0-101-generic #111-Ubuntu SMP Tue Mar 5 20:16:58 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / e08d0279c |
| Default Java | Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/branch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/branch-compile-tez-api-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/branch-compile-tez-dag-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/branch-compile-tez-api-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/branch-compile-tez-dag-jdkPrivateBuild-1.8.0_412-8u412-ga-1~22.04.1-b08.txt |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/buildtool-branch-checkstyle-tez-api.txt |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/buildtool-branch-checkstyle-tez-dag.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/branch-javadoc-tez-api-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/branch-javadoc-tez-dag-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/branch-findbugs-tez-api.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/branch-findbugs-tez-dag.txt |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/diff-checkstyle-tez-dag.txt |
| javadoc | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/artifact/out/diff-javadoc-javadoc-tez-api-jdkUbuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/testReport/ |
| Max. process+thread count | 431 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/3/console |
| versions | git=2.34.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 15s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 28s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m 39s |  master passed  |
| +1 :green_heart: |  compile  |   1m 24s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   1m 17s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  checkstyle  |   1m 30s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 16s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   1m  3s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +0 :ok: |  spotbugs  |   1m 23s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  9s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 54s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  javac  |   0m 54s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  findbugs  |   2m 36s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 22s |  tez-api in the patch passed.  |
| -1 :x: |  unit  |   5m 18s |  tez-dag in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   |  44m 58s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | tez.dag.app.TestSpeculation |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/362 |
| JIRA Issue | TEZ-4554 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 1489a2a5eaa0 5.15.0-101-generic #111-Ubuntu SMP Tue Mar 5 20:16:58 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / e08d0279c |
| Default Java | Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/4/artifact/out/patch-unit-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/4/testReport/ |
| Max. process+thread count | 372 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/4/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 16s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m  8s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  13m 57s |  master passed  |
| +1 :green_heart: |  compile  |   1m 21s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   1m 14s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  checkstyle  |   1m 35s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 17s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   1m  3s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +0 :ok: |  spotbugs  |   2m  9s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| -1 :x: |  findbugs  |   0m 31s |  tez-api in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 58s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   0m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  javac  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 36s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  findbugs  |   2m 28s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 25s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 20s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 27s |  The patch does not generate ASF License warnings.  |
|  |   |  46m 21s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/362 |
| JIRA Issue | TEZ-4554 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 9c1f6736304f 5.15.0-101-generic #111-Ubuntu SMP Tue Mar 5 20:16:58 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / e08d0279c |
| Default Java | Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/5/artifact/out/branch-findbugs-tez-api.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/5/testReport/ |
| Max. process+thread count | 373 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/5/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

ayushtkn commented on tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/DAGImpl.java:
```diff
@@ -2579,11 +2586,16 @@ void stopVertexServices() {
+  public void addUsedContainer(Container container) {
+    containersUsedByCurrentDAG.add(container.getId());
+    nodesUsedByCurrentDAG.add(container.getNodeId());
+    nodeHostsUsedByCurrentDAG.add(container.getNodeId().getHost());
```
can ``container.getNodeId()`` be ``null`` in any scenario?

abstractdog replied:
hm, should not be, in that case it's a yarn bug

ayushtkn replied:
Ohhk

ayushtkn reviewed (approved):
LGTM

abstractdog commented:
thanks @ayushtkn  for the review
I think that empty findbugs alert doesn't make any sense, I'll rerun the precommit test once again

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  23m 44s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 27s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  12m 49s |  master passed  |
| +1 :green_heart: |  compile  |   1m 20s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   1m 16s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  checkstyle  |   1m 27s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 17s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   1m  4s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +0 :ok: |  spotbugs  |   1m 19s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 57s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 54s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   0m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 48s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  javac  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  findbugs  |   2m 15s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 18s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 11s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 25s |  The patch does not generate ASF License warnings.  |
|  |   |  68m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/362 |
| JIRA Issue | TEZ-4554 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 42947a7313e1 5.15.0-106-generic #116-Ubuntu SMP Wed Apr 17 09:17:56 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 0ac505b09 |
| Default Java | Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/6/testReport/ |
| Max. process+thread count | 710 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/6/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

abstractdog commented:
seems like I have to slightly change this patch, because it returns all nodes by TaskSchedulerManager.getNumClusterNodes, which is served by a cachedNodeCount, local to the TaskSchedulerManager, it won't be accessible from the LlapTaskSchedulerService
I'm about to change this method to call and accumulate the individual TaskSchedulers' (e.g. the LLAP one) getClusterNodeCount to make LLAP able to be return with a real value:
https://github.com/apache/hive/blob/1c9969a003b09abc851ae7e19631ad208d3b6066/llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java#L1062

abstractdog commented:
with the new [commit](https://github.com/apache/tez/pull/362/commits/bb499355bb89ee81f99340216430b10bae901798), I was able to handle LLAP as well, updated the PR description
also removed NODE_HOSTS_USED_COUNT I felt a bit useless to handle it separately

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 14s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m  3s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  10m 54s |  master passed  |
| +1 :green_heart: |  compile  |   1m 21s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   1m 15s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  checkstyle  |   1m 30s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 15s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   1m  4s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +0 :ok: |  spotbugs  |   1m 20s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 57s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 47s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  javac  |   0m 47s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 24s |  tez-dag: The patch generated 1 new + 176 unchanged - 0 fixed = 177 total (was 176)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  findbugs  |   2m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 17s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 14s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 25s |  The patch does not generate ASF License warnings.  |
|  |   |  42m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/362 |
| JIRA Issue | TEZ-4554 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux d199429d8fac 5.15.0-106-generic #116-Ubuntu SMP Wed Apr 17 09:17:56 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 0ac505b09 |
| Default Java | Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/7/artifact/out/diff-checkstyle-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/7/testReport/ |
| Max. process+thread count | 691 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/7/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 14s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 20s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m 29s |  master passed  |
| +1 :green_heart: |  compile  |   1m 21s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   1m 16s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  checkstyle  |   1m 34s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 19s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   1m  4s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +0 :ok: |  spotbugs  |   1m 19s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  3s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 47s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  javac  |   0m 47s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 33s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  findbugs  |   2m 12s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 19s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 13s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   |  43m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/362 |
| JIRA Issue | TEZ-4554 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux b7763092e429 5.15.0-106-generic #116-Ubuntu SMP Wed Apr 17 09:17:56 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 0ac505b09 |
| Default Java | Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/8/testReport/ |
| Max. process+thread count | 438 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/8/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  22m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 47s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  10m  2s |  master passed  |
| +1 :green_heart: |  compile  |   1m 22s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   1m 15s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  checkstyle  |   1m 29s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 15s |  master passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   1m  6s |  master passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +0 :ok: |  spotbugs  |   1m 20s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 59s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 48s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  javac  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed with JDK Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed with JDK Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08  |
| +1 :green_heart: |  findbugs  |   2m 16s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 19s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 13s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   |  63m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/362 |
| JIRA Issue | TEZ-4554 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 7f89bbf01fa4 5.15.0-106-generic #116-Ubuntu SMP Wed Apr 17 09:17:56 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 0ac505b09 |
| Default Java | Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.23+9-post-Ubuntu-1ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_412-8u412-ga-1~22.04.1-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/9/testReport/ |
| Max. process+thread count | 367 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-362/9/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

ayushtkn reviewed (approved):
LGTM
